Doc/library/getopt.rst

```diff
@@ -144,13 +139,25 @@ In a script, typical usage is something like this::
                output = a
            else:
                assert False, "unhandled option"
-       # ...
+       process(args, output=output, verbose=verbose)
 
    if __name__ == "__main__":
        main()
 
 Note that an equivalent command line interface could be produced with less code
-and more informative help and error messages by using the :mod:`argparse` module::
+and more informative help and error messages by using the :mod:`optparse` module::
+
+   import optparse
+
+   if __name__ == '__main__':
+       parser = optparse.OptionParser()
+       parser.add_option('-o', '--output')
+       parser.add_option('-v', dest='verbose', action='store_true')
+       opts, args = parser.parse_args()
+       process(args, output=opts.output, verbose=opts.verbose)
+
+An equivalent command line interface for this simple case can also be produced
```

Comment:
Even in this simple example this is not an exact equivalent. For example, `-o -v` and `-o --` do not work, `-o=foo` works differently, you cannot interleave options and positional arguments. I said this is "a roughly equivalent".

Reply:
I think if we elaborate on the discrepancies here, it may help lay the foundation for weakening the current recommendation to use argparse in a future PR. I'll incorporate more of your notes here tomorrow.

Reply:
This topic now has a dedicated section at the start of the optparse docs (referenced from the argparse and getopt docs).